packages/theme/src/theme.ts

```diff
@@ -176,26 +209,25 @@ export class Theme extends HTMLElement {
         } else if (supportsAdoptingStyleSheets && this.shadowRoot) {
             const styleSheets = [];
             for (const style of styles) {
-                if (style.styleSheet) {
-                    styleSheets.push(style.styleSheet);
-                }
+                styleSheets.push(style.styleSheet as CSSStyleSheet);
             }
             this.shadowRoot.adoptedStyleSheets = styleSheets;
-        } else {
-            if (this.shadowRoot) {
-                const styleNodes = this.shadowRoot.querySelectorAll('style');
-                if (styleNodes) {
-                    styleNodes.forEach((element) => element.remove());
-                }
-                styles.forEach((s) => {
-                    const style = document.createElement('style');
-                    style.textContent = s.cssText;
-                    if (this.shadowRoot) {
-                        this.shadowRoot.appendChild(style);
-                    }
-                });
+        } else if (this.shadowRoot) {
```

Comment:
Pretty sure this next section was lifted straight from LitElement, but reading it as part of this change it seemed to make better sense with some simplified TS structures.